bittensor/utils/version.py

```diff
@@ -0,0 +1,74 @@
+from typing import Optional
+from pathlib import Path
+import time
+from packaging.version import Version
+
+import bittensor
+import requests
+
+VERSION_CHECK_THRESHOLD = 86400
+
+
+def _get_version_file_path() -> Path:
+    return Path.home() / ".bittensor" / ".last_known_version"
+
+
+def _get_version_from_file(version_file: Path) -> Optional[str]:
+    if not version_file.exists():
+        bittensor.logging.debug("No bitensor version file found")
+        return None
+
+    mtime = version_file.stat().st_mtime
+    bittensor.logging.debug(f"Found version file, last modified: {mtime}")
+    diff = time.time() - mtime
+
+    if diff >= VERSION_CHECK_THRESHOLD:
+        bittensor.logging.debug("Version file expired")
+        return None
+
+    try:
+        return version_file.read_text()
+    except OSError:
+        bittensor.logging.exception("Failed to read version file")
+        return None
```

Comment:
`version_file.exists()` is extra filesystem hit for no reason in here
additionally there is race condition between this .exists() and .stat() call below.
Is it likely to ever error out? No, but if we can get raceproof, simpler and slightly faster code - lets do that instead:)

```suggestion
    try:
        mtime = version_file.stat().st_mtime
        bittensor.logging.debug(f"Found version file, last modified: {mtime}")
            
        diff = time.time() - mtime
        if diff >= VERSION_CHECK_THRESHOLD:
            bittensor.logging.debug("Version file expired")
            return None
        
        return version_file.read_text()
    except FileNotFoundError:
        bittensor.logging.debug("No bitensor version file found")
        return None
    except OSError:
        bittensor.logging.exception("Failed to read version file")
        return None
```